apps/server/src/modules/board/uc/base.uc.ts

```diff
@@ -0,0 +1,50 @@
+import { AnyBoardDo, EntityId, SubmissionItem, UserRoleEnum } from '@shared/domain';
+import { ForbiddenException } from '@nestjs/common';
+import { AuthorizationService, Action } from '@modules/authorization';
+import { BoardDoAuthorizableService } from '../service';
+
+export abstract class BaseUc {
```

Comment:
In case of 5 or more checks it start making sense to factor out code blocks.
But the authorization is under current development. I would not change that code for now.
If you really want to factor out the code use a simple function. Like what we are doing for isFileElement or the mappers.

Easy to change. The base class is a much bigger layer addition which is not justified for the volatile  checkPermission method.

Reply:
It might make more sense as a utility class or composable than as an abstract superclass.
Im always hesitant about unnecessary inheritance structures as they complicate things without being very transparent.